prometheus/promhttp/http.go

```diff
@@ -406,3 +418,49 @@ func httpError(rsp http.ResponseWriter, err error) {
 		http.StatusInternalServerError,
 	)
 }
+
+func GetWriter(r *http.Request, rsp http.ResponseWriter, disableCompression bool, offeredCompressions []Compression) (io.Writer, error) {
+	w := io.Writer(rsp)
+	rsp.Header().Set(contentEncodingHeader, "identity")
```

Comment:
This is done by caller, could be deleted?

Reply:
I might wanna re-use this function within kube-state-metrics (currently uses the acceptEncoding function but copy'n'pasted). I made it work more nicely using the string enum.

Reply:
Refactored it to make rsp side-effect free. This is not done by the caller. :)